### PR TITLE
Publish snapshots to maven via GHA

### PR DESCRIPTION
**Pull request by lezzago**
*Issue #, if available:*
#802

*Description of changes:*
Publish snapshots to maven via GHA

*Testing:*
```
find alerting/snapshots | sort
alerting/snapshots
alerting/snapshots/org
alerting/snapshots/org/opensearch
alerting/snapshots/org/opensearch/plugin
alerting/snapshots/org/opensearch/plugin/alerting
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.pom
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.pom.md5
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.pom.sha1
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.pom.sha256
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.pom.sha512
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.zip
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.zip.md5
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.zip.sha1
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.zip.sha256
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/alerting-3.0.0.0-20230221.175611-1.zip.sha512
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/maven-metadata.xml
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/maven-metadata.xml.md5
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/maven-metadata.xml.sha1
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/maven-metadata.xml.sha256
alerting/snapshots/org/opensearch/plugin/alerting/3.0.0.0-SNAPSHOT/maven-metadata.xml.sha512
alerting/snapshots/org/opensearch/plugin/alerting/maven-metadata.xml
alerting/snapshots/org/opensearch/plugin/alerting/maven-metadata.xml.md5
alerting/snapshots/org/opensearch/plugin/alerting/maven-metadata.xml.sha1
alerting/snapshots/org/opensearch/plugin/alerting/maven-metadata.xml.sha256
alerting/snapshots/org/opensearch/plugin/alerting/maven-metadata.xml.sha512
```

*CheckList:*
[X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/maven-publish.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          2.*
+    ]
+
+jobs:
+  build-and-publish-snapshots:
+    strategy:
+      fail-fast: false
+      matrix:
+        jdk: [11, 17]
+        platform: ["ubuntu-latest", "windows-latest", "macos-latest"]
```
Comment (gaiksaya) [inline .github/workflows/maven-publish.yml:18]: I dont think we need to build on all platforms to publish to maven. Just ubuntu-latest works. Same with jdk.

**Review by gaiksaya — COMMENTED** [review]
Can you post the out of all the publications in the comment just to confirm? Testing information is provided in the issue
Thanks!

File: .github/workflows/maven-publish.yml (hunk 1 above)
Comment (lezzago) [reply .github/workflows/maven-publish.yml]: done

**Review by gaiksaya — APPROVED** [review]
LGTM! Added the secret role.

**Review by getsaurabh02 — APPROVED** [review]

**Pull request merged**